OsmAnd/src/net/osmand/plus/myplaces/AvailableGPXFragment.java

```diff
@@ -732,18 +732,55 @@ public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
 			@Override
 			public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
 				if (selectedItems.isEmpty()) {
-					Toast.makeText(getActivity(),
-							app.getString(R.string.local_index_no_items_to_do, actionButton.toLowerCase()),
-							Toast.LENGTH_SHORT).show();
+					app.showShortToastMessage(app.getString(R.string.local_index_no_items_to_do, actionButton.toLowerCase()));
 					return true;
 				}
+				if (actionResId == R.string.shared_string_delete){
+					AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+					builder.setMessage(getString(R.string.local_index_action_do, actionButton.toLowerCase(),
+							String.valueOf(selectedItems.size())));
+					builder.setPositiveButton(actionButton, listener);
+					builder.setNegativeButton(R.string.shared_string_cancel, null);
+					builder.show();
+				} else if (actionResId == R.string.local_index_mi_upload_gpx){
+					List<SelectableItem> allTracks = new ArrayList<>();
+					List<SelectableItem> selectedTracks = new ArrayList<>();
+					for (GpxInfo gpxInfo: selectedItems) {
+						SelectableItem s = createSelectableItem(gpxInfo);
```

Comment:
Strange method, some part of SelectableItem you create in createSelectableItem, other after it, why?
also setObject used two times